Sources/xcresultparser/CoberturaCoverageConverter.swift

```diff
@@ -0,0 +1,202 @@
+                lineElement.name = "line"
+                lineElement.addAttribute(XMLNode.nodeAttribute(withName: "number", stringValue: "\(line.lineNumber)"))
+                lineElement.addAttribute(XMLNode.nodeAttribute(withName: "branch", stringValue: "false"))
+
+                lineElement.addAttribute(XMLNode.nodeAttribute(withName: "hits", stringValue: "\(line.coverage)"))
+                linesElement.addChild(lineElement)
+            }
+        }
+
+        return doc.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
+    }
+    
+    private func fileCoverage(for file: String, relativeTo projectRoot: String) throws -> FileInfo {
+        let coverageData = try coverageForFile(path: file)
+        var file = FileInfo(path: relativePath(for: file, relativeTo: projectRoot), lines: [])
+        let pattern = #"(\d+):\s*(\d)"#
+        let regex = try NSRegularExpression(pattern: pattern, options: .anchorsMatchLines)
+        let nsrange = NSRange(coverageData.startIndex..<coverageData.endIndex,
+                              in: coverageData)
+        regex.enumerateMatches(in: coverageData, options: [], range: nsrange) { match, flags, stop in
+            guard let match = match else { return }
+            
+            let lineNumber = coverageData.text(in: match.range(at: 1))
+            let coverage = coverageData.text(in: match.range(at: 2))
+            
+            let line = LineInfo(lineNumber: lineNumber, coverage: Int(coverage) ?? 0)
+            file.lines.append(line)
+        }
+        return file
+    }
+    
+    func makeRootElement() -> XMLElement {
+        // TODO some of these values are B.S. - figure out how to calculate, or better to omit if we don't know?
+        let timeStamp = Date().timeIntervalSince1970
+        let rootElement = XMLElement(name: "coverage")
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "line-rate", stringValue: "\(codeCoverage.lineCoverage)"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "branch-rate", stringValue: "1.0"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "lines-covered", stringValue: "\(codeCoverage.coveredLines)"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "lines-valid", stringValue: "\(codeCoverage.executableLines)"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "timestamp", stringValue: "\(timeStamp)"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "version", stringValue: "diff_coverage 0.1"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "complexity", stringValue: "0.0"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "branches-valid", stringValue: "1.0"))
+        rootElement.addAttribute(XMLNode.nodeAttribute(withName: "branches-covered", stringValue: "1.0"))
+
+        return rootElement
+    }
+}
+
+struct LineInfo {
+    var lineNumber: String
```

Comment:
Any reason for these properties being 'var' instead of 'let'?

Reply:
That's a good point, I will try making them immutable.